src/vivarium_gates_mncnh/components/antenatal_care.py

```diff
@@ -285,38 +119,91 @@ def get_d_transition_probability(index: pd.Index) -> pd.Series:
             result[~is_full_term] = 1 - ancfirst[~is_full_term]
             return result
 
-        # Decisions
-        # Initial state transitions to A, B, C, or D
-        initial_state.add_transition(
-            output_state=state_A,
-            probability_function=get_a_transition_probability,
+        # this ordering is important when using propensities
+        # to preserve correlations in facility choice model
+        probabilities = pd.concat(
+            [
+                get_no_visit_probability(event.index),
+                get_later_visit_only_probability(event.index),
+                get_early_visit_only_probability(event.index),
+                get_both_visits_probability(event.index),
+            ],
+            axis=1,
         )
-        initial_state.add_transition(
-            output_state=state_B,
-            probability_function=get_b_transition_probability,
+        probabilities.columns = [
+            ANC_ATTENDANCE_TYPES.NONE,
+            ANC_ATTENDANCE_TYPES.LATER_PREGNANCY_ONLY,
+            ANC_ATTENDANCE_TYPES.FIRST_TRIMESTER_ONLY,
+            ANC_ATTENDANCE_TYPES.FIRST_TRIMESTER_AND_LATER_PREGNANCY,
+        ]
+
+        # use correlated propensity to decide ANC attendance
+        propensities = self.propensity(event.index)
+        anc_choices = probabilities.columns
+        anc_attendance = _choice(propensities, anc_choices, probabilities.values)
+        anc_attendance.name = COLUMNS.ANC_ATTENDANCE
+
+        self.population_view.update(anc_attendance)
+
+
+class Ultrasound(Component):
+    @property
+    def columns_created(self):
+        return [
+            COLUMNS.ULTRASOUND_TYPE,
+            COLUMNS.STATED_GESTATIONAL_AGE,
+        ]
+
+    @property
+    def columns_required(self):
+        return [COLUMNS.ANC_ATTENDANCE, COLUMNS.GESTATIONAL_AGE_EXPOSURE]
+
+    def setup(self, builder: Builder):
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.location = get_location(builder)
+        self.scenario = INTERVENTION_SCENARIOS[builder.configuration.intervention.scenario]
+
+    def build_all_lookup_tables(self, builder: Builder) -> None:
```

Comment:
You don't need to overwrite this method. You can just put this in `configuration_defaults`.